apiv2/db_import/common/finders.py

```diff
@@ -0,0 +1,45 @@
+
+import logging
+import re
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Any
+
+from db_import.common.config import DBImportConfig
+
+from platformics.database.models.base import Base
+
+logger = logging.getLogger("db_import")
+
+if TYPE_CHECKING:
+    from db_import.importers.base import ItemDBImporter
+else:
+    ItemDBImporter = Any
+
+
+class ItemFinder(ABC):
+    @abstractmethod
+    def __init__(self, config: DBImportConfig, **kwargs):
+        pass
+
+    @abstractmethod
+    def find(self, item_importer: ItemDBImporter) -> list[ItemDBImporter]:
+        pass
+
+
+class FileFinder(ItemFinder):
```

Comment:
We'll need to add more finders over time, but this is ~similar to the idea of finders in s3 ingestion.